AAI-179: initialize FastAPI project with basic structure and tests (#2)
AustralianBioCommons/aai-backend: README.md

# aai-backend

The backend library between Auth0 and the AAI portal front end. Technical design document for this project lives in [this Google doc](https://docs.google.com/document/d/1W3-7Hme08M-b4kwMvcQoUscznVNxtOldxuKYPPRhBFE/edit?tab=t.0).

## Installation

Follow these steps to set up the project:

1. **Clone the Repository**:

   ```bash
   git clone https://github.com/AustralianBioCommons/aai-backend.git
   cd aai-backend
   ```

2. **Set Up a Virtual Environment**:
   Create and activate a virtual environment:

   ```bash
   python3 -m venv .venv
   source .venv/bin/activate
   ```

3. **Install Dependencies**:
   Install the required dependencies:

   ```bash

>    pip install "fastapi[standard]"

Reviewer: Suggest to the installation of python packages into a requirements.txt file:
1. add a new requirements.txt file
2. populate this file with the packages (ex: `fastapi[standard]`, `pytest` and `httpx`
3. update the instructions instead to be `pip install requirements.txt -r`

Reviewer 2: If we are using `uv` then we should use `uv` to manage the requirements - we can add new dependencies via `uv add <dependency>` and they will be added to the dependencies in `pyproject.toml`. For packages we only need during dev/testing we can do `uv add --dev pytest`.

   pip install pytest httpx
   ```

> 4. **Run the Application**:

Reviewer: ```suggestion
## Run the Application
```

Since the previous section is "Installation".

   Use `uvicorn` to run the FastAPI application:

   ```bash
   fastapi dev main.py
   ```

> 5. **Run Tests**:

Reviewer: ```suggestion
Run Tests
```

Since the previous section is "Installation" and "Run Tests"

   Execute the test suite using `pytest`:
   ```bash
   pytest
   ```

## Notes

> - Ensure you have Python 3.13 or higher installed.

Reviewer: ```
Ensure you have Python 3.13 or higher installed.
``` 

This should be moved to another section at the beginning of this README as "Requirements"

> - Use the virtual environment for all development to avoid dependency conflicts.

Reviewer: Can remove this line since we are already suggesting to use `venv`.

# AAI Backend

The backend library between Auth0 and the AAI portal front end. Technical design document for this project lives in [this Google doc](https://docs.google.com/document/d/1W3-7Hme08M-b4kwMvcQoUscznVNxtOldxuKYPPRhBFE/edit?tab=t.0).

## Requirements

> Ensure you have Python 3.13 or higher installed.

Reviewer: ```suggestion
Ensure you have the following installed:
- python 3.13+
- uv ([installation instructions](https://github.com/astral-sh/uv#installation))
```

## Installation

Follow these steps to set up the project:

1. **Clone the Repository**:

   ```bash
   git clone https://github.com/AustralianBioCommons/aai-backend.git
   cd aai-backend
   ```

2. **Set Up a Virtual Environment**:
   Create and activate a virtual environment:

   ```bash
   uv venv
   source .venv/bin/activate
   ```

3. **Install Dependencies**:
   Install the required dependencies:

   ```bash

>    uv sync

Reviewer: ```suggestion
 (aai-backend)  uv sync
```

Reviewer: Adding this hint so that users know to do this step in the venv

>    uv add -r requirements.txt

Reviewer 2: If we are using `uv` we shouldn't need a `requirements.txt`. You can:

* Delete `requirements.txt`
* Change this instruction to just `uv sync`
